File: grida-core/src/main/java/org/grida/datetime/DateTimePicker.java
package org.grida.datetime;

import java.time.LocalDateTime;

public class DateTimePicker {

    public LocalDateTime now() {
        return LocalDateTime.now();
    }

    public DateTimeRange thisMonthRange() {
        LocalDateTime now = LocalDateTime.now();

        return new DateTimeRange(
                now.withDayOfMonth(1),
                now.withDayOfMonth(now.toLocalDate().lengthOfMonth())
        );
    }

    public DateTimeRange monthOfYearRange(int year, int month) {
        LocalDateTime startDate = LocalDateTime.of(year, month, 1, 0, 0);

        return new DateTimeRange(
                startDate.withDayOfMonth(1),
                startDate.withDayOfMonth(startDate.toLocalDate().lengthOfMonth())
        );
    }
}

File: grida-common/src/main/kotlin/org/grida/datetime/DateTimePicker.kt
package org.grida.datetime

import java.time.LocalDateTime
import java.time.ZoneId
import java.time.ZonedDateTime

class DateTimePicker {

    fun now(): LocalDateTime {
        return ZonedDateTime.now(ZoneId.of(TIME_ZONE_SEOUL)).toLocalDateTime()
    }

    fun monthOfYearRange(
        year: Int,
        month: Int
    ): DateTimeRange {
        val targetDate = ZonedDateTime.of(year, month, 1, 0, 0, 0, 0, ZoneId.of(TIME_ZONE_SEOUL))
            .toLocalDateTime()

        return DateTimeRange(
            targetDate.withDayOfMonth(1),
            targetDate.withDayOfMonth(targetDate.toLocalDate().lengthOfMonth())
        )
    }

    companion object Util {

        private val dateTimePicker = DateTimePicker()

        fun now(): LocalDateTime {
            return dateTimePicker.now()
        }

        fun monthOfYearRange(
            year: Int,
            month: Int
        ): DateTimeRange {
            return dateTimePicker.monthOfYearRange(year, month)
        }
    }
}
